kore/src/Kore/Logger/Registry.hs

```diff
@@ -0,0 +1,164 @@
+{- |
+Copyright   : (c) Runtime Verification, 2019
+License     : NCSA
+-}
+
+module Kore.Logger.Registry
+    ( parseEntryType
+    , toSomeEntryType
+    , lookupTextFromTypeWithError
+    , registry
+    , typeToText
+    , textToType
+    , getEntryTypesAsText
+    -- entry types
+    , debugAppliedRuleType
+    , debugAxiomEvaluationType
+    , debugSolverSendType
+    , debugSolverRecvType
+    , warnBottomHookType
+    , warnFunctionWithoutEvaluatorsType
+    , warnSimplificationWithRemainderType
+    ) where
+
+import Control.Applicative
+    ( empty
+    )
+import Data.Functor.Classes
+    ( eq2
+    )
+import Data.Map.Strict
+    ( Map
+    )
+import qualified Data.Map.Strict as Map
+import Data.Text
+    ( Text
+    )
+import qualified Data.Text as Text
+import Data.Tuple
+    ( swap
+    )
+import Data.Typeable
+    ( Proxy (..)
+    )
+import qualified Text.Megaparsec as Parser
+import Type.Reflection
+    ( SomeTypeRep (..)
+    , someTypeRep
+    , typeOf
+    )
+
+import Kore.Logger
+    ( Entry
+    )
+import Kore.Logger.DebugAppliedRule
+    ( DebugAppliedRule
+    )
+import Kore.Logger.DebugAxiomEvaluation
+    ( DebugAxiomEvaluation
+    )
+import Kore.Logger.DebugSolver
+    ( DebugSolverRecv
+    , DebugSolverSend
+    )
+import Kore.Logger.WarnBottomHook
+    ( WarnBottomHook
+    )
+import Kore.Logger.WarnFunctionWithoutEvaluators
+    ( WarnFunctionWithoutEvaluators
+    )
+import Kore.Logger.WarnSimplificationWithRemainder
+    ( WarnSimplificationWithRemainder
+    )
+
+data Registry =
+    Registry
+    { textToType :: Map Text SomeTypeRep
+    , typeToText :: Map SomeTypeRep Text
```

Comment:
Why do we need to store `typeToText`?

Reply:
We need to print the entry type information when logging (see https://github.com/ana-pantilie/kore/blob/447988ddb59f8c0ec0a2b22bdb1877d36fd5f36d/kore/src/Kore/Logger/Output.hs#L337).